src/UnkObjects/unk8600F8.cpp

```diff
@@ -25,3 +26,201 @@ void unk_8600F8::loc_611DC0() {
+            entry = entry->next;
+        }
+    }
+
+    if (this->unk02 != 0) {
+        this->entries[index] = sub_611DF0(a, b, this->entries[index]);
+        this->num_entries++;
+        return true;
+    }
+
+    hash_map_entry *newEntry = (hash_map_entry*)MC2_MALLOC(sizeof(hash_map_entry));
+    if (newEntry != nullptr) {
+        newEntry->name = MC2_STRDUP(a);
+        newEntry->value = b;
+        newEntry->next = this->entries[index];
+    }
+
+    this->entries[index] = newEntry;
+    this->num_entries++;
+
+    if (this->num_entries <= this->max_entries) {
+        return true;
+    }
+    
+    if (glo_679778 == 0)
+        return true;
+
+    uint32_t esi_store = glo_8602C8;
+    glo_8602C8 = 5;
+    // Increase num entry size?
+    sub_6125A0(this->max_entries * 4);
+
+    glo_8602C8 = esi_store;
+
+    return true;
+}
+
+bool unk_8600F8::sub_611FE0(uint32_t *eax) {
+    if (this->entries == nullptr)
+        return false;
+
+    hash_map_entry *entry = sub_611F60(0);
+
+    if (entry == nullptr)
+        return false;
+
+    eax[0] = 0;
+    eax[1] = (uint32_t)entry->name;
+    eax[2] = (uint32_t)entry->value;
+    return true;
+}
+
+hash_map_entry *unk_8600F8::sub_611F60(int32_t entryIndex) {
+    if (entryIndex == -1)
+        return nullptr;
+
+    int hashEntryIndex = -1;
+    int numFoundEntries = -1;
+    int edx = 0;
+    int esi = 0;
+    bool entryFound = false;
+    hash_map_entry *entry = nullptr;
+
+    hash_map_entry *lastFound = nullptr;
+
+    do {
+        if (hashEntryIndex >= (int)this->max_entries)
+            return nullptr;
+        if (entryFound) {
+            entryFound = false;
+
+            entry = entry->next;
+
+            if (entry == nullptr) {
+                continue;
+            }
+
+            if (entry == lastFound) {
+                continue;
+            }
+
+            entryFound = true;
+            numFoundEntries++;
+            continue;
+        }
+
+        if (hashEntryIndex >= (int)(this->max_entries - 1))
+            return nullptr;
+
+        entry = this->entries[++hashEntryIndex];
+
+        if (entry == nullptr)
+            continue;
+
+        entryFound = true;
+        lastFound = entry;
+        numFoundEntries++;
+    } while (entryIndex != numFoundEntries);
+
+    return entry;
+}
+
+bool unk_8600F8::sub_612020(uint32_t *eax) {
+    hash_map_entry *entry = sub_611F60(eax[0] + 1);
+
+    if (entry == nullptr)
+        return false;
+
+    eax[0]++;
+    eax[1] = (uint32_t)entry->name;
+    eax[2] = (uint32_t)entry->value;
+    return true;
+}
+
+bool sub_612E10(const char* key) {
+    unk_612150 *result = glo_8600F8.sub_6124A0(key);
+    return result != nullptr;
+}
+
+bool sub_612E30(const char *key, std::uint32_t index, int32_t * decimal_value) {
+    unk_612150 *entry = glo_8600F8.sub_6124A0(key);
+
+    if (entry == nullptr)
+        return false;
+
+    if (index >= entry->count)
+        return false;
+
+    *decimal_value = sub_61A5DC(entry->args[index]);
```

Comment:
Note to self: Investigate if `sub_61A5DC` is `atoi` or similar, and if so, replace it with `lexical_cast`.